cli/fabric.c

```diff
@@ -484,7 +485,9 @@ static int topo_info(int argc, char **argv)
 
 	argconfig_parse(argc, argv, CMD_DESC_TOPO_INFO, opts, &cfg, sizeof(cfg));
 
-	for(i = 0; i < SWITCHTEC_MAX_PORTS; i++)
+	max_ports = switchtec_max_supported_ports(cfg.dev);
+
+	for(i = 0; i < max_ports; i++)
```

Comment:
I don't really see the point of this patch. It's initializing a structure that's larger than the what the hardware might support and that's fine. It doesn't seem like it solves any actual bugs and just makes the code more complicated.

Reply:
For this commit, I was trying to remove the possibility of accessing beyond the max number of ports supported by the hardware. For `switchtec status` calls, reading beyond the max supported number of ports resulted in garbage and seg faults. If that's not an issue in this commit I can drop it.

Reply:
Yeah, the first two commits look good, I'd drop the last one.